### ISSUE TITLE
WSL2 DNS server blocks on PTR requests

### ISSUE THREAD

**Issue by lmoze** [microsoft/WSL#6157]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.572]
Your Distribution version: [Ubuntu-20.04]
Whether the issue is on WSL 2 and/or WSL 1: [WSL2 only]
```

# Steps to reproduce

* Ensure DNSutils are installed: `sudo apt-get install dnsutils`
* Start continuous DNS lookups: `while :; do date; dig microsoft.com a | grep 'Query time:'; sleep 0.1; echo; done`
* In another WSL2 console window, start a PTR lookup for any RFC1918 IP address: `dig 10.0.0.1 ptr`
* Note that the continuous DNS A record lookups stall until after the PTR lookup receives its result

**WSL logs**: 

#  Expected behavior

The WSL2 DNS server should continue to service other unrelated DNS requests while the PTR lookup is in progress

# Actual behavior

All other DNS lookups from inside WSL2 stall until after the PTR lookup completes.

# Additional Information

Solving our VPN DNS troubles is out of scope for this issue; this is included only as background info:

* The DNS resolution stall during PTR lookups became apparent while I was debugging DNS issues that WSL2 experiences on computers connected to our corporate Pritunl VPN. 
* Our VPN client's DNS setup is causing DNS PTR queries from inside WSL2 to take around 30 seconds to resolve. Because the DNS server fails to respond to other queries while the PTR query is in-progress, this means all WSL2 name resolution stalls for up to 30 seconds.
* The DNS stall can break `apt-get update` on networks with slow PTR resolution. apt sends out a PTR query at the start of the update process, and other DNS queries started in parallel to look up the upstream hosts may time out while the DNS server is blocked responding to the PTR request.
* These side-effects may be related to https://github.com/microsoft/WSL/issues/4246 - in particular, a comment on that issue noted the same `apt-get update` errors I experienced https://github.com/microsoft/WSL/issues/4246#issuecomment-691680847 

This issue can be partially worked around by adding a high DNS timeout to /etc/resolv.conf - e.g. `options timeout:30`
